### PR TITLE
RELATED: ONE-4768 LiveExamples extended by ThemeProvider

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**
JIRA: ONE-4768

Extend LiveExamples by ThemeProvider and some themed components example. Using externally provided theme instead of loading one from workspace as we don't want to use theming on whole workspace to keep current state intact

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [x] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1590/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3171/ : ABORTED in 10m 20s

File: libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx [hunk 1]
```diff
@@ -27,6 +27,17 @@ export interface IThemeProviderProps {
      * component tree.
      */
     workspace?: string;
+
+    /**
+     * Theme object to use instead of one loaded from workspace
+     */
+    adhocTheme?: ITheme;
```
Comment (lupko) [inline libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx:34]: I suggest to rename to just `theme`. Theme object to use. If not specified, provider will load theme from workspace.

File: libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx [hunk 2]
```diff
@@ -27,6 +27,17 @@ export interface IThemeProviderProps {
...
+
+    /**
+     * If true, global theme will be automatically cleared during component unmount.
+     * Default value is false
+     */
+    autoClear?: boolean;
```
Comment (lupko) [inline libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx:40]: do we need this? if the correct behavior is to always clear stuff after itself, then why allowing not to do it? 

it seems that it leaks impl detail - that there is something stateful about the theme.

would this autoClear make sense if we had just styled components all way through?

I suggest to remove this and always clear after itself..

File: libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx (hunk 1 above)
Comment (no23reason) [reply libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx:34]: Seconded, in
https://github.com/gooddata/gooddata-ui-sdk/pull/779/commits/bce80e5a9c9db928a8e135ce4a259b2378cac9c8
https://github.com/gooddata/gooddata-ui-sdk/pull/779/commits/4eec1f73d7fe48aa8bf131bc360543e117233362
I will be doing the same for the backend and workspace contexts (hopefully, that will get merged soon).

File: libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx (hunk 2 above)
Comment (no23reason) [inline libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx:40]: Should this be true by default? I mean why wouldn't you want this?

File: libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx (hunk 1 above)
Comment (lupko) [reply libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx:34]: also, there is an interplay with the backend/workspace and contexts. please expand the docs to mention the expectations

Comment (ivan-nejezchleb) [reply libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx:34]: @lupko Which doc do you mean? We have this mention of Backend/WorkspaceProvider above in this interface comments:
export interface IThemeProviderProps {
    /**
     * Analytical backend, from which the ThemeProvider will obtain selected theme object
     *
     * If you do not specify instance of analytical backend using this prop, then you MUST have
     * BackendProvider up in the component tree.
     */
    backend?: IAnalyticalBackend;
    /**
     * Identifier of analytical workspace, from which the ThemeProvider will obtain the selected theme identifier
     *
     * If you do not specify workspace identifier, then you MUST have WorkspaceProvider up in the
     * component tree.
     */
    workspace?: string;

File: libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx (hunk 2 above)
Comment (ivan-nejezchleb) [reply libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx:40]: removed and done by default

Comment (ivan-nejezchleb) [reply libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx:40]: removed and done by default

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1598/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3179/ : SUCCESS in 11m 56s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1600/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3181/ : SUCCESS in 13m 44s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1602/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3183/ : SUCCESS in 12m 10s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/513/ : SUCCESS in 25s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/773/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3184/ : SUCCESS in 11m 50s

**Pull request merged**